sound/soc/sof/loader.c

```diff
@@ -494,6 +582,20 @@ int snd_sof_load_firmware_raw(struct snd_sof_dev *sdev)
 
 	kfree(fw_filename);
 
+	/* check for extended manifest */
+	xman_size = snd_sof_xman_size(plat_data->fw->data);
+	if (xman_size > 0) {
+		ret = snd_sof_fw_xman_parse(sdev, plat_data->fw->data);
+
+		/* when no error occurred, drop extended manifest  */
+		if (ret > 0) {
+			ret = strip_fw(sdev, xman_size);
+		} else {
+			dev_err(sdev->dev, "error: request firmware %s failed when parsing extended manifest: %d\n",
+				fw_filename, ret);
```

Comment:
fw_filename is kfree'd line 583, can't work

Reply:
right